Storage/Rust/RustPlaces.swift

```diff
@@ -408,29 +408,48 @@ public class RustPlaces: BookmarksHandler, HistoryMetadataObserver {
 
 // MARK: History APIs
 
-extension VisitTransition {
-    public static func fromVisitType(visitType: VisitType) -> Self {
-        switch visitType {
-        case .unknown:
-            return VisitTransition.link
-        case .link:
-            return VisitTransition.link
-        case .typed:
-            return VisitTransition.typed
-        case .bookmark:
-            return VisitTransition.bookmark
-        case .embed:
-            return VisitTransition.embed
-        case .permanentRedirect:
-            return VisitTransition.redirectPermanent
-        case .temporaryRedirect:
-            return VisitTransition.redirectTemporary
-        case .download:
-            return VisitTransition.download
-        case .framedLink:
-            return VisitTransition.framedLink
-        case .recentlyClosed:
-            return VisitTransition.link
+// WKWebView has these:
+/*
+WKNavigationTypeLinkActivated,
+WKNavigationTypeFormSubmitted,
+WKNavigationTypeBackForward,
+WKNavigationTypeReload,
+WKNavigationTypeFormResubmitted,
+WKNavigationTypeOther = -1,
+*/
+
+// The previous VisitType started from .unknown at 1 which means everything is off
+// from desktop and rust by 1. 
+extension VisitType {
+    public static func fromRawValue(rawValue: Int?) -> Self {
+        switch rawValue {
+        case 2: return .link
```

Comment:
nit: Totally up to you, but I can't help wonder if these magic number should be replaced with enums. Something like 

```
enum VisitTypeValues: Int {
    case link = 2, typed, bookmark, . . .  
}
```

Might be more trouble than it's worth though.

Reply:
ooo thats a good idea, Though I wonder if it clears anything up since it's mostly an internal transition